src/components/outline-nav/components/outline-nav-with-active/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react'
+import OutlineNav from 'components/outline-nav'
+import { OutlineLinkItem } from 'components/outline-nav/types'
+import {
+	getItemSlugs,
+	highlightActiveItems,
+} from 'components/outline-nav/utils'
+import { useDeviceSize } from 'contexts'
+import { useActiveSection } from 'lib/hash-links/use-active-section'
+
+/**
+ * Given an array of OutlineLinkItems,
+ * render an outline nav, and automatically highlight the active item.
+ */
+function OutlineNavWithActive({
+	items,
+}: {
+	items: Omit<OutlineLinkItem, 'isActive'>[]
+}) {
+	/**
+	 * We useDeviceSize to avoid running active heading calculations
+	 * if we're on a viewport where we won't render the nav anyways.
+	 *
+	 * TODO: isDesktop isn't fully accurate, we may want to adjust this
+	 * based on the specific breakpoint at which we show the sidecar.
```

Comment:
💡 I remember some logic to work around this with the mobile menu that could be worth referencing.